src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostBuilderTests.cs

```diff
@@ -826,6 +826,160 @@ private class ServiceC
+            Assert.Equal("AppValue", configuration["TestKey"]); // Should come from base settings, not Production
+            Assert.Null(configuration["ProductionKey"]); // Should not be loaded from Production file
+        }
+
+        [Fact]
+        public void ConfigureDefaults_DoesNotLoadApplicationSpecificSettings_WhenApplicationNameIsEmpty()
+        {
+            using TempDirectory tempDir = new();
+            string appSettingsPath = Path.Combine(tempDir.Path, ".settings.json");
+
+            // Create test configuration file that should NOT be loaded
+            File.WriteAllText(appSettingsPath, """{"TestKey": "ShouldNotBeLoaded"}""");
+
+            using var host = new HostBuilder()
+                .ConfigureDefaults(args: null)
+                .UseContentRoot(tempDir.Path)
+                .ConfigureHostConfiguration(config =>
+                {
+                    config.AddInMemoryCollection(new[]
+                    {
+                        new KeyValuePair<string, string>(HostDefaults.ApplicationKey, ""),
+                        new KeyValuePair<string, string>(HostDefaults.EnvironmentKey, "Production")
+                    });
+                })
+                .Build();
+
+            var configuration = host.Services.GetRequiredService<IConfiguration>();
+
+            // Verify that app-specific settings are not loaded when ApplicationName is empty
+            Assert.Null(configuration["TestKey"]);
+        }
+
+        [Fact]
+        public void ConfigureDefaults_ReplacesPathSeparatorsInApplicationName()
+        {
+            using TempDirectory tempDir = new();
+            string appSettingsPath = Path.Combine(tempDir.Path, "my_app.settings.json");
+
+            // Create test configuration file with path separators replaced by underscores
+            File.WriteAllText(appSettingsPath, """{"TestKey": "PathSeparatorValue"}""");
+
+            using var host = new HostBuilder()
+                .ConfigureDefaults(args: null)
+                .UseContentRoot(tempDir.Path)
+                .ConfigureHostConfiguration(config =>
+                {
+                    config.AddInMemoryCollection(new[]
+                    {
+                        new KeyValuePair<string, string>(HostDefaults.ApplicationKey, "my/app"),
+                        new KeyValuePair<string, string>(HostDefaults.EnvironmentKey, "Production")
+                    });
+                })
+                .Build();
+
+            var configuration = host.Services.GetRequiredService<IConfiguration>();
+            var hostEnvironment = host.Services.GetRequiredService<IHostEnvironment>();
+
+            // Verify that ApplicationName retains original value (not sanitized)
+            Assert.Equal("my/app", hostEnvironment.ApplicationName);
+            
+            // Verify that path separators are replaced with underscores for file loading
```

Comment:
This comment could be better, e.g.:
```suggestion
            // Verify the value from the application specific settings file with the sanitized name is loaded
```